### PR TITLE
style: Remove trailing whitespace and missing final newlines

### PR DESCRIPTION
**Pull request by hyperupcall**
In #272, one section stated to "delete trailing spaces". This does that, and also missing final newlines (so files are files by POSIX)

This also integrates `editorconfig-checker` with GitHub Actions so the style will continue to remain consistent. It runs only on files that have changed for speed. I used `editorconfig-checker` because integrating `shfmt` might have changed more lines than what might be considered necessary; `editorconfig-checker` checks the bare minimum.

The only real code change resulting from this is in [tools/uninstall.sh](https://github.com/ohmybash/oh-my-bash/compare/master...hyperupcall:whitespace?expand=1#diff-e51454ebff6b9e7d3db470db8412ccb7a569b916fd4d7f7e997a079355f26cdeR24)

### CODE REVIEW

File: tools/uninstall.sh [hunk 1]
```diff
@@ -20,8 +21,9 @@ _omb_uninstall_find_bashrc_original() {
     break
   done
   unset _omb_uninstall_file
```
Comment (akinomyoga) [inline tools/uninstall.sh:23]: I think it is a good change. Could you also add `-v` here?

**Review by akinomyoga — COMMENTED** [review]
Thank you for your work!

I actually left these tasks in #272 to avoid possible conflicts with existing PRs, but I think now it is time to actively work on them. When I took over the maintenance of this project, there were about 50 or more pending PRs, which I hadn't time to look into. But we have already processed most of the important PRs that existed at that time.

----

Maybe we do not have to decide it in this PR, but another thing that I might need to rethink is the format of the commit message. There were really no explicit rules for the format of the commit message, but as for the first line, I was loosely following the suggestion made in [CONTRIBUTING.md](https://github.com/ohmybash/oh-my-bash/blob/789a13c6aace11eb41847a19e9534ccec8996189/CONTRIBUTING.md) of #90.

> ```
> <FILE>: <Summary>
>
> <Description>
> <Fixes/Bug>: #<BUG_NUMBER>
> Signed-off-by: <name> <surname> <e-mail>
> ```

However, I actually have no plan to merge #90. Maybe I would pick some changes from #90, but I probably reject most of the changes suggested in #90.

Also, personally, I'd like to switch to Conventional Commits as in this PR. What do you think?

**Comment by akinomyoga** [comment]
@hyperupcall

File: tools/uninstall.sh [hunk 2]
```diff
@@ -11,6 +11,7 @@ _omb_uninstall_contains_omb() {
 _omb_uninstall_find_bashrc_original() {
   _omb_uninstall_bashrc_original=
   printf '%s\n' "Looking for original bash config..."
+  _omb_old_ofs=$IFS
```
[the remark this answers is not in the document]
Comment (akinomyoga) [reply tools/uninstall.sh]: 55bbdef

File: tools/uninstall.sh [hunk 3]
```diff
@@ -20,8 +21,9 @@ _omb_uninstall_find_bashrc_original() {
     break
   done
   unset _omb_uninstall_file
-  IFS=' 	
-'
+  IFS=$_omb_old_ofs
```
[the remark this answers is not in the document]
Comment (akinomyoga) [reply tools/uninstall.sh]: 55bbdeff75ac38f049ad258ac6e53dddfb36c514

File: tools/uninstall.sh (hunk 1 above)
Comment (akinomyoga) [reply tools/uninstall.sh]: 55bbdeff75ac38f049ad258ac6e53dddfb36c514

File: .git-blame-ignore-revs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# style: Remove trailing whitespace and add missing final newlines
+5e2d79d3de414100443b76bdc173e2229f617b5a
```
[the remark this answers is not in the document]
Comment (hyperupcall) [reply .git-blame-ignore-revs]: I think I rebased the original style commit and forgot to update the `.git-blame-ignore-revs` file

[the remark this answers is not in the document]
Comment (akinomyoga) [reply .git-blame-ignore-revs]: Thank you for your reply! There was no reply soon, so I have modified the mentioned parts by myself. Could you check these changes?

**Comment by hyperupcall** [comment]
> I actually left these tasks in https://github.com/ohmybash/oh-my-bash/issues/272 to avoid possible conflicts with existing PRs, but I think now it is time to actively work on them. When I took over the maintenance of this project, there were about 50 or more pending PRs, which I hadn't time to look into. But we have already processed most of the important PRs that existed at that time.

:+1: 

> Also, personally, I'd like to switch to Conventional Commits as in this PR. What do you think?

I think Conventional Commits is good to use - it's what is used at [asdf](https://github.com/asdf-vm/asdf) too. I don't have any strong opinions about it though

I do apologize for my late reply, I know I started work on this a week ago

File: .git-blame-ignore-revs (hunk 4 above)
[the remark this answers is not in the document]
Comment (hyperupcall) [reply .git-blame-ignore-revs]: I'll pull these changes right now :+1:

File: tools/uninstall.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (hyperupcall) [reply tools/uninstall.sh]: I did not think about this case - thank you for clarifying!

**Comment by hyperupcall** [comment]
I looked things over and the new changes look good to me! I'll look over #432 tomorrow morning

**Comment by akinomyoga** [comment]
Thanks!

**Pull request merged**